File: DziennikElektroniczny/Models/Grade.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Data;
using System;
using System.Collections.Generic;

namespace DziennikElektroniczny.Models
{
    public class Grade
    {
        [Key]
        public int GradeId { get; set; }

        [Range(1, 1000000000)]
        public int? StudentPersonId { get; set; }

        [Range(1, 1000000000)]
        public int? TeacherPersonId { get; set; }

        public virtual Person StudentPerson { get; set; }
        public virtual Person TeacherPerson { get; set; }

        [Range(1, 1000000000)]
        public int SubjectId { get; set; }
        public Subject Subject { get; set; }

        [Required]
        [Column(TypeName = "varchar(4)")]
        public string Value { get; set; }

        [Required]
        [DataType(DataType.Date)]
        public DateTime Date { get; set; }

    }
}

File: DziennikElektroniczny/AngularApp/frontend/src/app/models/Grade.ts
export class GradeViewModel {
  constructor(
    public id: number,
    public studentPersonId: number,
    public studentDisplayName: string,
    public teacherPersonId: number,
    public teacherDisplayName: string,
    public subjectId: number,
    public subjectName: string,
    public value: string,
    public date: Date,
  ) {}
}
